### ISSUE TITLE
After splitting my dataset into train/eval/test splits with CsvExampleGen, StatisticsGen fails to create stats for test set

### ISSUE THREAD

**Issue by jackhawa** [tensorflow/tfx#924]
I was able to split my dataset sucessfully with CsvExampleGen and produce 3 artifacts:

![image](https://user-images.githubusercontent.com/7749809/68607056-c9b92a80-047d-11ea-9f1d-a5f8551a4721.png)

But StatisticsGen seems to only allow producing stats for train and eval as can be seen with this line (in artifact.py): 

![image](https://user-images.githubusercontent.com/7749809/68607098-e190ae80-047d-11ea-9818-a9c299df508d.png)

The error I get is: 0 elements matches split test

![image](https://user-images.githubusercontent.com/7749809/68607162-fec57d00-047d-11ea-9b47-d4573ee9c4ec.png)

I believe the fix is to add the string "test" to DEFAULT_EXAMPLE_SPLITS.

**Comment by jackhawa** [comment]
Anyway this fix can be made on tfx version 1.14.0?

**Comment by numerology** [comment]
I think another way to fix this is by manually specifying output when constructing StatisticsGen component.

Something like 
```python
output=types.Channel(
        type=standard_artifacts.ExampleStatistics,
        artifacts=[
            standard_artifacts.ExampleStatistics(split=split)
            for split in ['train', 'eval', 'test']
        ])
```

**Comment by 1025KB** [comment]
The downstream components currently doesn't support splits other than train and eval yet, and yes changing DEFAULT_EXAMPLE_SPLITS would make statistics gen works, but other downstream components would still ignore test split as train and eval is hardcoded in most downstream components

**Comment by Ark-kun** [comment]
> The downstream components currently doesn't support splits other than train and eval yet, and yes changing DEFAULT_EXAMPLE_SPLITS would make statistics gen works, but other downstream components would still ignore test split as train and eval is hardcoded in most downstream components

I think this might be a design issue.
Is there any effort to deprecate `split` as a property of every artifact?
I think that in most cases a split dataset is analogous to subdirectories of a dataset directory which groups them.

**Comment by 1025KB** [comment]
Currently single Channel contains multiple artifacts, each artifact is represent a split. we are working on make it merged, single channel will contains one artifacts which represent multiple splits.

Add Charles for more details

**Comment by tkakantousis** [comment]
I ran into the same issue with [Taxi Beam Pipeline](https://github.com/tensorflow/tfx/blob/0.14.0/tfx/examples/chicago_taxi_pipeline/taxi_pipeline_beam.py) 
 (Python 2.7, TFX 0.14) when using a mysql database for metadata. It works ok with mysqlite.

**Comment by charlesccychen** [comment]
@Ark-kun Yes, we will be doing this refactoring to remove split from all artifacts soon, ideally in the next release.  Specifically, in the next release, the output of ExampleGen will be a single artifact with multiple subdirectory splits and we will work to allow subsequent components to process these properly (note that this will be a breaking change).  While the workaround above by @numerology works for the current release, it will stop working in the future.

**Comment by jiyongjung0** [comment]
We don't have any restriction in split names now. Let me close the issue. Feel free to re-open if needed.
